MaxText/maxtext_utils.py

```diff
@@ -124,3 +126,53 @@ def calculate_tflops_prefill(num_model_parameters, prefill_length, config, log=T
           f'\t\tCausal attention TFLOPs: {causal_attention_tflops} ',
           f'({100 * causal_attention_tflops/total_tflops:.2f})% of Total')
   return total_tflops, learnable_weight_tflops, causal_attention_tflops
+
+
+def calc_not_sharded_params(shard,
+                      expected_per_device_num_param,
+                      key,
+                      pspec,
+                      num_devices_sharded
+                      ):
+  """Returns the number of params that aren't sharded."""
+  new_num_p = np.prod(shard.data.shape)
+
+  if new_num_p == expected_per_device_num_param:
+    print(f'Input is sharded over {num_devices_sharded} devices as expected!')
+  else:
+    print(f'Expected {expected_per_device_num_param} params but got {new_num_p}')
+    print(f'Off by a factor of {new_num_p // expected_per_device_num_param}')
+    print('key path', key)
+    print('pspec', pspec)
+    if jax.process_index() == 0:
+      return new_num_p
+  return 0
+
+def is_sharded_correctly(params):
```

Comment:
Let's just call this whatever we end up wanting more descriptively (instead of `is_sharded_correctly`)

Probably assert_params_sufficiently_sharded(params, mesh, tol=1.01)

Reply:
thanks, changed naming.